infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -418,27 +338,67 @@ private void handleTxToken(CorfuPayloadMsg<TokenRequest> msg,
         final long serverEpoch = r.getServerEpoch();
         final TokenRequest req = msg.getPayload();
+        final TxResolutionInfo txInfo = req.getTxnResolution();
+        final long txSnapshotTimestamp = txInfo.getSnapshotTimestamp();
 
-        // Since Java does not allow an easy way for a function to return multiple values, this
-        // variable is passed to the consumer that will use it to indicate to us if/what key was
-        // responsible for an aborted transaction.
-        AtomicReference<Integer> conflictKey = new AtomicReference(TokenResponse.NO_CONFLICT_KEY);
-
-        // in the TK_TX request type, the sequencer is utilized for transaction conflict-resolution.
-        // Token allocation is conditioned on commit.
-        // First, we check if the transaction can commit.
-        TokenType tokenType = txnCanCommit(req.getTxnResolution(), conflictKey);
-        if (tokenType != TokenType.NORMAL) {
-            // If the txn aborts, then DO NOT hand out a token.
-            Token token = new Token(Address.ABORTED, serverEpoch);
-            r.sendResponse(ctx, msg, CorfuMsgType.TOKEN_RES.payloadMsg(new TokenResponse(tokenType,
-                    conflictKey.get(), token, Collections.emptyMap())));
+        // The sequencer is a failover sequencer and we have no information,
+        // so the request fails.
+        if (txSnapshotTimestamp < globalLogStart.get() - 1) {
+            log.debug("handleTxToken: ABORT[{}] snapshot-ts[{}] failover-ts[{}]",
+                    txSnapshotTimestamp, globalLogStart.get());
+            r.sendResponse(ctx, msg, CorfuMsgType.TOKEN_RES.payloadMsg(
+                    new TokenResponse(TokenType.TX_ABORT_NEWSEQ, Address.ABORTED, serverEpoch)));
             return;
         }
 
+        // Iterate over all of the streams in the conflict set
+        for (Map.Entry<UUID, Set<Integer>> entry : txInfo.getConflictSet().entrySet()) {
+            final Set<Integer> conflictParamSet = entry.getValue();
+            final UUID streamId = entry.getKey();
+
+            // if conflict-parameters are present, check for conflict
+            // based on conflict-parameter updates
+            if (conflictParamSet != null && conflictParamSet.size() > 0) {
+                // for each key pair, check for conflict;
+                // if not present, check against the wildcard
+                conflictParamSet.forEach(conflictParam -> {
+                    final int conflictKeyHash = getConflictHashCode(streamId, conflictParam);
+                    final Long conflictTail = conflictToGlobalTailCache
+                            .getIfPresent(conflictKeyHash);
+
+                    if (conflictTail != null
+                            && (conflictTail > txSnapshotTimestamp
+                            || maxConflictWildcard > txSnapshotTimestamp)) {
+                        log.debug("handleTxToken: ABORT[{}] conflict-key[{}]({}ts={})", txInfo,
```

Comment:
I think for the benefit of the system user, if we can keep it at warn for sometime it would be much better. We will be able to root cause aborts better.